include/fiction/algorithms/simulation/sidb/critical_temperature.hpp

```diff
@@ -165,188 +140,137 @@ template <typename Lyt>
 class critical_temperature_impl
 {
   public:
-    critical_temperature_impl(const Lyt& lyt, const critical_temperature_params& params,
+    critical_temperature_impl(const Lyt& lyt, const critical_temperature_params& ps,
                               critical_temperature_stats<Lyt>& st) :
             layout{lyt},
-            parameter{params},
-            temperature_stats{st}
-    {}
+            params{ps},
+            stats{st},
+            bii(bdl_input_iterator<Lyt>{layout, params.bdl_params})
 
-    bool gate_based_simulation()
+    {
+        stats.critical_temperature = params.max_temperature;
+    }
+
+    /**
+     * *Gate-based Critical Temperature* Simulation of a SiDB layout for a given Boolean function.
+     *
+     * @tparam TT The type of the truth table specifying the gate behavior.
+     * @param tt Expected Boolean function of the layout given as a multi-output truth table.
+     * @return `True` if the simulation succeeds and *Gate-based Critical Temperature* is determined, `false` otherwise.
+     */
+    template <typename TT>
+    bool gate_based_simulation(const std::vector<TT>& tt) noexcept
```

Comment:
```suggestion
    [[nodiscard]] bool gate_based_simulation(const std::vector<TT>& spec) noexcept
```

The name `tt` is used in the `fiction` namespace already.